ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,387 @@
+            MOI.FEASIBILITY_SENSE,
+            MOI.Nonlinear.Model(),
+            Dict{String,Any}(),
+        )
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.SolverVersion) = "0.4.37"
+
+function MOI.is_empty(model::Optimizer)
+    # TODO replace `isnothing(model.nlp_model.objective)`
+    #      by `MOI.is_empty` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    return isnothing(model.manifold) &&
+           isempty(model.variable_primal_start) &&
+           isnothing(model.nlp_model.objective) &&
+           model.sense == MOI.FEASIBILITY_SENSE
+end
+
+function MOI.empty!(model::Optimizer)
+    model.manifold = nothing
+    model.problem = nothing
+    model.state = nothing
+    empty!(model.variable_primal_start)
+    model.sense = MOI.FEASIBILITY_SENSE
+    # TODO replace by `MOI.empty!` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    model.nlp_model.objective = nothing
+    return nothing
+end
+
+"""
+    MOI.supports(::Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Return a `Bool` indicating whether `attr.name` is a valid option name
+for `Manopt`.
+"""
+function MOI.supports(::Optimizer, ::MOI.RawOptimizerAttribute)
+    # FIXME It should depend on `attr.name`
+    return true
+end
+
+function MOI.get(model::Optimizer, attr::MOI.RawOptimizerAttribute)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
+    return model.options[attr.name]
+end
+
+function MOI.set(model::Optimizer, attr::MOI.RawOptimizerAttribute, value)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
+    model.options[attr.name] = value
+    return nothing
+end
+
+MOI.get(::Optimizer, ::MOI.SolverName) = "Manopt"
+
+"""
+    MOI.supports_incremental_interface(::Optimizer)
+
+Return `true` indicating that `Manopt.Optimizer` implements
+`MOI.add_constrained_variables` and `MOI.set` for
+`MOI.ObjectiveFunction` so it can be used with `JuMP.direct_model`
+and does not require a `MOI.Utilities.CachingOptimizer`.
+"""
+MOI.supports_incremental_interface(::Optimizer) = true
+
+"""
+    MOI.copy_to(dest::Optimizer, src::MOI.ModelLike)
+
+Because `supports_incremental_interface(dest)` is `true` we can simply
+use `MOI.Utilities.default_copy_to`.
+"""
+function MOI.copy_to(dest::Optimizer, src::MOI.ModelLike)
+    return MOI.Utilities.default_copy_to(dest, src)
+end
+
+"""
+    MOI.supports_add_constrained_variables(::Optimizer, ::Type{<:VectorizedManifold})
+
+Return `true` indicating that `Manopt.Optimizer` support optimization on
+variables constrained to belong in a vectorized manifold.
+"""
+function MOI.supports_add_constrained_variables(::Optimizer, ::Type{<:VectorizedManifold})
+    return true
+end
+
+"""
+    MOI.add_constrained_variables(model::Optimizer, set::VectorizedManifold)
+
+Add `MOI.dimension(set)` variables constrained in `set` and return the list
```

Comment:
We could mention here that `set` is a `VectorizedManifold` and link to the corresponding type?